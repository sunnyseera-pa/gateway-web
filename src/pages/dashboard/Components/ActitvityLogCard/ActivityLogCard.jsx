/** @jsx jsx */
import { jsx } from '@emotion/react';
import React from 'react';
import { Row, Col } from 'react-bootstrap';
import PropTypes from 'prop-types';
import SLA from '../../../commonComponents/sla/SLA';
import DatasetOnboardingHelper from '../../../../utils/DatasetOnboardingHelper.util';
import { dateFormats } from '../../../../utils/GeneralHelper.util';
import * as styles from './ActivityLogCard.styles';

const ActivityLogCard = props => {
	const { versionNumber, applicationStatus, dateSubmitted, events } = props;
	return (
		<Row>
			<div className='col-md-12'>
				<div className='layoutCard'>
					<div css={styles.activityCard}>
						<Row css={styles.activityLog}>
							<Col sm={6} lg={6} data-testid='version-title'>
								<h1>{`Version ${versionNumber}`}</h1>
								<span>{`Submitted ${dateFormats(dateSubmitted).dateOnly}`}</span>
							</Col>
							<Col sm={6} lg={6}>
								<span css={styles.applicationStatus} data-testid='status'>
									<SLA
										classProperty={DatasetOnboardingHelper.datasetStatusColours[applicationStatus]}
										text={DatasetOnboardingHelper.datasetSLAText[applicationStatus]}
									/>
								</span>
							</Col>
						</Row>
						{events.map((event, i) => {
							const timestamp = dateFormats(event.timestamp);
							return (
								<div key={`event-${i}`}>
									<Row css={styles.activityLog}>
										<Col sm={12} lg={12}>
											<h1>
												<span data-testid={`event-title-${i}`}>{timestamp.dateOnly}</span>
											</h1>
										</Col>
									</Row>
									<Row css={styles.activityLog}>
										<Col sm={1} lg={1}>
											<span>{timestamp.timeOnly}</span>
										</Col>
										<Col sm={1} lg={1}>
											<span data-testid={`event-time-${i}`}>{timestamp.timeOnly}</span>
										</Col>
										<Col sm={10} lg={10}>
											<span data-testid={`event-text-${i}`}>{event.plainText}</span>
										</Col>
									</Row>
									{event.detailedText && (
										<Row css={styles.activityLog}>
											<Col sm={1} lg={1}></Col>
											<Col sm={1} lg={1}></Col>
											<Col sm={10} lg={10}>
												<div className='jumbotron' data-testid={`event-detailed-text-${i}`}>
													<span>{event.detailedText}</span>
												</div>
											</Col>
										</Row>
									)}
								</div>
							);
						})}
					</div>
				</div>
			</div>
		</Row>
	);
};

ActivityLogCard.propTypes = {
	applicationStatus: PropTypes.string.isRequired,
	versionNumber: PropTypes.number.isRequired,
	dateSubmitted: PropTypes.string.isRequired,
	events: PropTypes.arrayOf(PropTypes.object),
};

export default ActivityLogCard;
