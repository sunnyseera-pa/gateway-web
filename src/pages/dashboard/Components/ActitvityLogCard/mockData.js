export default {
	version: 'Version 2.0.0',
	versionNumber: 2,
	dateSubmitted: '2021-11-03T14:18:45.536Z',
	applicationStatus: 'active',
	events: [
		{
			_id: '61829a9ae4ea0f9763bd3fd7',
			userTypes: ['applicant', 'custodian'],
			eventType: 'datasetVersionApproved',
			logType: 'dataset',
			timestamp: '2021-11-03T14:20:10.420Z',
			plainText: 'Version 2.0.0 of this dataset has been approved by portal admin Callum Reekie',
			html: '<b>Version 2.0.0</b> of this dataset has been approved by portal admin <b>Callum Reekie</b>',
			detailedText: 'Comment:\nThis is a quality dataset!',
			detailedHtml:
				"<div class='activity-log-detail'><div class='activity-log-detail-header'>Comment:</div><div class='activity-log-detail-row'>This is a quality dataset!</div></div>",
			user: '611cfb5a19d737331dad4aff',
			version: '2.0.0',
			versionId: '61829a39e4ea0f9763bd3f6b',
			__v: 0,
		},
		{
			_id: '61829a45e4ea0f9763bd3f8c',
			userTypes: ['applicant', 'custodian'],
			eventType: 'newDatasetVersionSubmitted',
			logType: 'dataset',
			timestamp: '2021-11-03T14:18:45.576Z',
			plainText: 'Version 2.0.0 of this dataset has been submitted by data custodian Callum Reekie',
			html: '<b>Version 2.0.0</b> of this dataset has been submitted by data custodian <b>Callum Reekie</b>',
			user: '616993c3034a7d773064e208',
			version: '2.0.0',
			versionId: '61829a39e4ea0f9763bd3f6b',
			__v: 0,
		},
	],
};
