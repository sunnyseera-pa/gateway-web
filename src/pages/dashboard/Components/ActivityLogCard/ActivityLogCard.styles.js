import { css } from '@emotion/react';

export const activityCard = css`
	padding: 20px 30px 15px 25px;
	font-size: 14px;
`;

export const applicationStatus = css`
	float: right;
`;

export const activityLog = css`
	margin-bottom: 20px;
`;
