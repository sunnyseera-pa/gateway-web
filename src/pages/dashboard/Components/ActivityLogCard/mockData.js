export default {
	version: 'Version 2.0.0',
	versionNumber: 2,
	applicationStatus: 'active',
	dateSubmitted: '2021-11-08T14:49:41.225Z',
	meta: {
		dateSubmitted: '2021-11-08T14:49:41.225Z',
		dateCreated: '2021-11-08T14:39:43.513Z',
		applicationStatus: 'active',
	},
	events: [
		{
			_id: '61893936df1feefcdd2b7939',
			eventType: 'datasetVersionApproved',
			logType: 'dataset',
			timestamp: '2021-11-08T14:50:30.646Z',
			user: '611cfb5a19d737331dad4aff',
			userDetails: {
				firstName: 'Callum',
				lastName: 'Reekie',
				role: 'admin',
			},
			version: '2.0.0',
			versionId: '618936afe87de1fbd16d1135',
			userTypes: ['admin', 'custodian'],
			adminComment: 'Approved with minor changes!',
			__v: 0,
		},
		{
			_id: '61893936df1feefcdd2b792c',
			eventType: 'datasetUpdatesSubmitted',
			logType: 'dataset',
			timestamp: '2021-11-08T14:50:30.566Z',
			user: '611cfb5a19d737331dad4aff',
			userDetails: {
				firstName: 'Callum',
				lastName: 'Reekie',
				role: 'admin',
			},
			version: '2.0.0',
			versionId: '618936afe87de1fbd16d1135',
			userTypes: ['admin', 'custodian'],
			datasetUpdates: [
				{
					'summary/abstract': {
						previousAnswer: 'test',
						updatedAnswer: 'I have changed the abstract!',
					},
				},
				{
					'provenance/temporal/distributionReleaseDate': {
						previousAnswer: '',
						updatedAnswer: '02/11/2021',
					},
				},
			],
			__v: 0,
		},
		{
			_id: '61893905df1feefcdd2b78b8',
			eventType: 'newDatasetVersionSubmitted',
			logType: 'dataset',
			timestamp: '2021-11-08T14:49:41.380Z',
			user: '616993c3034a7d773064e208',
			userDetails: {
				firstName: 'Callum',
				lastName: 'Reekie',
				role: 'custodian',
			},
			version: '2.0.0',
			versionId: '618936afe87de1fbd16d1135',
			userTypes: ['admin', 'custodian'],
			__v: 0,
		},
	],
};
