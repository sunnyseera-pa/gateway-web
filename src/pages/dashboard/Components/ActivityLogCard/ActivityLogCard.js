/** @jsx jsx */
import { jsx } from '@emotion/react';
import React, { Suspense } from 'react';
import { useTranslation } from 'react-i18next';
import { Row, Col } from 'react-bootstrap';
import PropTypes from 'prop-types';
import SLA from '../../../commonComponents/sla/SLA';
import ListInfo from '../../../commonComponents/ListInfo';
import DatasetOnboardingHelper from '../../../../utils/DatasetOnboardingHelper.util';
import { dateFormats } from '../../../../utils/GeneralHelper.util';
import '../../../../i18n';
import tick from '../../../../images/tick.svg';
import amber from '../../../../images/attention.svg';
import rejected from '../../../../images/Application_rejected.svg';
import * as styles from './ActivityLogCard.styles';
import Timeline from '../../../commonComponents/Timeline/Timeline';
import BlockQuote from '../../../commonComponents/Blockquote';
import _ from 'lodash';

let eventStatusIcons = {
	newDatasetVersionSubmitted: rejected,
	datasetVersionApproved: tick,
	datasetVersionRejected: rejected,
	datasetVersionArchived: rejected,
	datasetVersionUnarchived: rejected,
	datasetUpdatesSubmitted: rejected,
};

const ActivityLogCard = props => {
	const { t } = useTranslation();
	const {
		versionNumber,
		meta: { applicationStatus },
		dateSubmitted,
		events,
	} = props;

	return (
		<Suspense fallback={t('loading')}>
			<Row>
				<div className='col-md-12'>
					<div className='layoutCard'>
						<div css={styles.activityCard}>
							<Row css={styles.activityLog}>
								<Col sm={6} lg={6} data-testid='version-title'>
									<h1>{`Version ${versionNumber}`}</h1>
									<span className='gray800-14'>{t('datetime.submitted', { datetime: dateFormats(dateSubmitted).dateOnly })}</span>
								</Col>
								<Col sm={6} lg={6}>
									<span css={styles.applicationStatus} data-testid='status'>
										<SLA
											classProperty={DatasetOnboardingHelper.datasetStatusColours[applicationStatus]}
											text={DatasetOnboardingHelper.datasetSLAText[applicationStatus]}
										/>
									</span>
								</Col>
							</Row>
							<Timeline
								data={events.map((event, i) => {
									const dateTime = dateFormats(event.timestamp);
									const commentTitle = event.eventType === 'datasetVersionRejected' ? t('activitylog.rejection') : t('comment');

									return {
										icon: <img src={eventStatusIcons[event.eventType]} data-testid={`${i}-${event.eventType}`} alt='Icon' />,
										time: dateTime.timeOnly,
										content: (
											<div>
												<div
													dangerouslySetInnerHTML={{
														__html: t(event.eventType, { versionNumber: event.version.replace(/\..*$/, ''), ...event.userDetails }),
													}}
													className='mb-3'
												/>
												{event.adminComment && (
													<BlockQuote>
														<h6 className='mb-3 gray800'>{commentTitle}</h6>
														<p className='mb-0'>{event.adminComment}</p>
													</BlockQuote>
												)}
												{event.datasetUpdates &&
													event.datasetUpdates.map(updates => {
														const key = Object.keys(updates)[0];

														return (
															<BlockQuote data-testid={`event-detailed-text-${i}`}>
																<h6 className='mb-3 gray800'>{_.startCase(key.replace(/[^\/]*$/g, '')).replace(/\s/g, ' | ')}</h6>
																<ListInfo
																	data={[
																		{
																			label: t('question'),
																			value: _.startCase(key.replace(/^.*\//, '')),
																		},
																		{
																			label: t('previousAnswer'),
																			value: updates[key].previousAnswer,
																		},
																		{
																			label: t('updatedAnswer'),
																			value: updates[key].updatedAnswer,
																		},
																	]}
																/>
															</BlockQuote>
														);
													})}
											</div>
										),
									};
								})}
							/>
						</div>
					</div>
				</div>
			</Row>
		</Suspense>
	);
};

ActivityLogCard.propTypes = {
	applicationStatus: PropTypes.string.isRequired,
	versionNumber: PropTypes.number.isRequired,
	meta: {
		dateSubmitted: PropTypes.string.isRequired,
		dateCreated: PropTypes.string.isRequired,
		applicationStatus: PropTypes.string.isRequired,
	},
	events: PropTypes.arrayOf(PropTypes.object),
};

export default ActivityLogCard;
