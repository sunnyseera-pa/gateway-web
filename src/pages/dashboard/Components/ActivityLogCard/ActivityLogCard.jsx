/** @jsx jsx */
import { jsx } from '@emotion/react';
import PropTypes from 'prop-types';
import { Suspense } from 'react';
import { Col, Row } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';
import ACTIVITY_LOG_PROP_TYPES from '../../../../services/activitylog/activitylog';
import DatasetOnboardingHelper from '../../../../utils/DatasetOnboardingHelper.util';
import { dateFormats } from '../../../../utils/GeneralHelper.util';
import BlockQuote from '../../../storybookComponents/BlockQuote';
import Icon from '../../../storybookComponents/Icon';
import ListInfo from '../../../storybookComponents/ListInfo';
import SLA from '../../../commonComponents/sla/SLA';
import Timeline from '../../../storybookComponents/Timeline';
import * as styles from './ActivityLogCard.styles';

let eventStatusIcons = {
	newDatasetVersionSubmitted: 'Versions_created',
	datasetVersionApproved: 'Application_approved',
	datasetVersionRejected: 'Application_rejected',
	datasetVersionArchived: 'Updates_requested',
	datasetVersionUnarchived: 'Updates_requested',
	datasetUpdatesSubmitted: 'Updates_requested',
};

const ActivityLogCard = props => {
	const { t } = useTranslation();
	const {
		versionNumber,
		meta: { applicationStatus, dateSubmitted },
		events,
		mb,
	} = props;

	return (
		<Suspense fallback={t('loading')}>
			<Row>
				<div className='col-md-12'>
					<div className='layoutCard'>
						<div css={styles.activityCard}>
							<Row css={styles.activityLog(mb)}>
								<Col sm={6} lg={6} data-testid='version-title'>
									<h1>{`Version ${versionNumber}`}</h1>
									<span className='gray800-14'>{t('datetime.submitted', { datetime: dateFormats(dateSubmitted).dateOnly })}</span>
								</Col>
								<Col sm={6} lg={6}>
									<span css={styles.applicationStatus} data-testid='status'>
										<SLA
											classProperty={DatasetOnboardingHelper.datasetStatusColours[applicationStatus]}
											text={DatasetOnboardingHelper.datasetSLAText[applicationStatus]}
										/>
									</span>
								</Col>
							</Row>
							<Timeline
								data={events.map((event, index) => {
									const dateTime = dateFormats(event.timestamp);
									const commentTitle = event.eventType === 'datasetVersionRejected' ? t('activitylog.rejection') : t('comment');

									return {
										icon: <Icon name={eventStatusIcons[event.eventType]} data-testid={`${index}-${event.eventType}`} size='4xl' />,
										time: dateTime.timeOnly,
										content: (
											<div>
												<div
													dangerouslySetInnerHTML={{
														__html: t(event.eventType, { versionNumber: event.version.replace(/\..*$/, ''), ...event.userDetails }),
													}}
													className='mb-3'
												/>
												{event.adminComment && (
													<BlockQuote>
														<h6 className='mb-3 gray800'>{commentTitle}</h6>
														<p className='mb-0'>{event.adminComment}</p>
													</BlockQuote>
												)}
												{event.datasetUpdates &&
													event.datasetUpdates.map((item, i) => {
														const log = DatasetOnboardingHelper.getUpdatesSubmittedLog(item);

														return (
															<div data-testid={`event-detailed-text-${index}-${i}`}>
																<BlockQuote>
																	<h6 className='mb-3 gray800'>{log.heading}</h6>
																	<ListInfo
																		data={[
																			{
																				label: t('question'),
																				value: log.question,
																			},
																			{
																				label: t('previousAnswer'),
																				value: log.answers.previousAnswer,
																			},
																			{
																				label: t('updatedAnswer'),
																				value: log.answers.updatedAnswer,
																			},
																		]}
																	/>
																</BlockQuote>
															</div>
														);
													})}
											</div>
										),
									};
								})}
							/>
						</div>
					</div>
				</div>
			</Row>
		</Suspense>
	);
};

ActivityLogCard.defaultProps = {
	mb: 20,
};

ActivityLogCard.propTypes = {
	...ACTIVITY_LOG_PROP_TYPES,
	mb: PropTypes.string,
};

export default ActivityLogCard;
