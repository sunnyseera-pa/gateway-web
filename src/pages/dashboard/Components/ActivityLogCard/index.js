import ActivityLogCard from './ActivityLogCard';

export default ActivityLogCard;
