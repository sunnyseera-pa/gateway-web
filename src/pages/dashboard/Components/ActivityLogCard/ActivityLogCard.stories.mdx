import { Meta, Story, Canvas } from '@storybook/addon-docs';
import ActivityLogCard from './ActivityLogCard';
import mockData from './mockData';

<Meta title='ActivityLogCard' component={ActivityLogCard} args={mockData} />

<Canvas>
	<Story name='Basic'>{props => <ActivityLogCard {...props} />}</Story>
</Canvas>
