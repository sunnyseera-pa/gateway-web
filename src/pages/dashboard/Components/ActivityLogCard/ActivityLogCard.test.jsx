import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { QueryClient } from 'react-query';
import DatasetOnboardingHelper from '../../../../utils/DatasetOnboardingHelper.util';
import { dateFormats } from '../../../../utils/GeneralHelper.util';
import ActivityLogCard from './ActivityLogCard';
import mockData from './mockData';

const props = {
	...mockData,
};
let wrapper;

describe('Given the ActivityLogCard component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<ActivityLogCard {...props} />, {
				wrapper: Providers,
			});
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then  Title should be rendered with  Submitted Date', () => {
			expect(screen.getByTestId(`version-title`)).toHaveTextContent(`Version ${props.versionNumber}`);
		});
		it('Then  Status should be rendered', () => {
			const statusText = DatasetOnboardingHelper.datasetSLAText[props.meta.applicationStatus];

			expect(screen.getByTestId('status')).toHaveTextContent(statusText);
		});

		it('Then  all events should be rendered', () => {
			props.events.map((event, i) => {
				const timestamp = dateFormats(event.timestamp);
				expect(screen.getByTestId(`event-time-${i}`)).toHaveTextContent(timestamp.timeOnly);
			});
		});

		it('Then updates submiited log be rendered', () => {
			props.events.map((event, index) => {
				if (event.datasetUpdates) {
					event.datasetUpdates.map((item, i) => {
						const log = DatasetOnboardingHelper.getUpdatesSubmittedLog(item);
						const text = `${log.heading}question${log.question}previousAnswer${log.answers.previousAnswer}updatedAnswer${log.answers.updatedAnswer}`;
						expect(screen.getByTestId(`event-detailed-text-${index}-${i}`)).toHaveTextContent(text);
					});
				}
			});
		});
	});
});
