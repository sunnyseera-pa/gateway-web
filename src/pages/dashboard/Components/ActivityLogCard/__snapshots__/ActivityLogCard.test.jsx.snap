// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ActivityLogCard component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-md-12"
    >
      <div
        class="layoutCard"
      >
        <div
          class="css-1rfjxm6-ActivityLogCard"
        >
          <div
            class="css-1s3f5pl-ActivityLogCard row"
          >
            <div
              class="col-lg-6 col-sm-6"
              data-testid="version-title"
            >
              <h1>
                Version 2
              </h1>
              <span
                class="gray800-14"
              >
                datetime.submitted
              </span>
            </div>
            <div
              class="col-lg-6 col-sm-6"
            >
              <span
                class="css-zlbsue-ActivityLogCard"
                data-testid="status"
              >
                <div
                  class="sla sla-green"
                >
                  Live
                </div>
              </span>
            </div>
          </div>
          <ul
            class="ui-Timeline css-1sivg29 css-1rag44v-Timeline"
          >
            <li
              class="css-f5erw7"
            >
              <div
                class="css-z67izj"
              >
                <div
                  class="timeline-icon css-1vyp46c"
                >
                  <div />
                </div>
                <div
                  class="css-1pfaqp5"
                >
                  <time
                    data-testid="event-time-0"
                    datetime="14:50"
                  >
                    14:50
                  </time>
                </div>
              </div>
              <div
                class="css-obendf"
              >
                <div>
                  <div
                    class="mb-3"
                  >
                    datasetVersionApproved
                  </div>
                  <blockquote
                    class="ui-Blockquote css-1sivg29 css-1ta6g32-BlockQuote"
                  >
                    <h6
                      class="mb-3 gray800"
                    >
                      comment
                    </h6>
                    <p
                      class="mb-0"
                    >
                      Approved with minor changes!
                    </p>
                  </blockquote>
                </div>
              </div>
            </li>
            <li
              class="css-f5erw7"
            >
              <div
                class="css-z67izj"
              >
                <div
                  class="timeline-icon css-1vyp46c"
                >
                  <div />
                </div>
                <div
                  class="css-1pfaqp5"
                >
                  <time
                    data-testid="event-time-1"
                    datetime="14:50"
                  >
                    14:50
                  </time>
                </div>
              </div>
              <div
                class="css-obendf"
              >
                <div>
                  <div
                    class="mb-3"
                  >
                    datasetUpdatesSubmitted
                  </div>
                  <div
                    data-testid="event-detailed-text-1-0"
                  >
                    <blockquote
                      class="ui-Blockquote css-1sivg29 css-1ta6g32-BlockQuote"
                    >
                      <h6
                        class="mb-3 gray800"
                      >
                        Summary
                      </h6>
                      <ul
                        class="ui-ListInfo css-1sivg29 css-1jupbku-ListInfo"
                      >
                        <li
                          class="css-oc8kcp"
                        >
                          <div
                            class="css-112bk1i"
                          >
                            question
                          </div>
                          <div>
                            Abstract
                          </div>
                        </li>
                        <li
                          class="css-oc8kcp"
                        >
                          <div
                            class="css-112bk1i"
                          >
                            previousAnswer
                          </div>
                          <div>
                            test
                          </div>
                        </li>
                        <li
                          class="css-oc8kcp"
                        >
                          <div
                            class="css-112bk1i"
                          >
                            updatedAnswer
                          </div>
                          <div>
                            I have changed the abstract!
                          </div>
                        </li>
                      </ul>
                    </blockquote>
                  </div>
                  <div
                    data-testid="event-detailed-text-1-1"
                  >
                    <blockquote
                      class="ui-Blockquote css-1sivg29 css-1ta6g32-BlockQuote"
                    >
                      <h6
                        class="mb-3 gray800"
                      >
                        Provenance | Temporal
                      </h6>
                      <ul
                        class="ui-ListInfo css-1sivg29 css-1jupbku-ListInfo"
                      >
                        <li
                          class="css-oc8kcp"
                        >
                          <div
                            class="css-112bk1i"
                          >
                            question
                          </div>
                          <div>
                            Distribution Release Date
                          </div>
                        </li>
                        <li
                          class="css-oc8kcp"
                        >
                          <div
                            class="css-112bk1i"
                          >
                            previousAnswer
                          </div>
                          <div />
                        </li>
                        <li
                          class="css-oc8kcp"
                        >
                          <div
                            class="css-112bk1i"
                          >
                            updatedAnswer
                          </div>
                          <div>
                            02/11/2021
                          </div>
                        </li>
                      </ul>
                    </blockquote>
                  </div>
                </div>
              </div>
            </li>
            <li
              class="css-f5erw7"
            >
              <div
                class="css-z67izj"
              >
                <div
                  class="timeline-icon css-1vyp46c"
                >
                  <div />
                </div>
                <div
                  class="css-1pfaqp5"
                >
                  <time
                    data-testid="event-time-2"
                    datetime="14:49"
                  >
                    14:49
                  </time>
                </div>
              </div>
              <div
                class="css-obendf"
              >
                <div>
                  <div
                    class="mb-3"
                  >
                    newDatasetVersionSubmitted
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
